src/fileActions.jsx

```diff
@@ -436,16 +438,18 @@ const EditPermissionsModal = ({ selected, path }) => {
     );
 };
 
-const setDirectoryName = (val, setName, setNameError, setErrorMessage) => {
+const setDirectoryName = (val, setName, setNameError, setErrorMessage, cwdInfo) => {
     setErrorMessage(undefined);
     setName(val);
 
     if (val === "") {
-        setNameError(_("Directory name cannot be empty."));
+        setNameError(_("Name cannot be empty."));
     } else if (val.length >= 256) {
-        setNameError(_("Directory name too long."));
+        setNameError(_("Name too long."));
     } else if (val.includes("/")) {
-        setNameError(_("Directory name cannot include a /."));
+        setNameError(_("Name cannot include a /."));
+    } else if (cwdInfo?.entries[val]) {
```

Comment:
So no "Rename and overwrite" option, at present?

Reply:
Not presently, I have supported it in `renameat2` but not here as this was a completely new scenario and when I made UI for it I did not receive positive feedback.